qtop.py

```diff
@@ -1368,88 +1390,283 @@ def check_python_version():
+                if not options.YAML_EXISTS:
+                    convert_to_yaml(scheduler, INPUT_FNs_commands, input_filenames)
+                yaml_files = get_yaml_files(scheduler, input_filenames)
+
+                worker_nodes = get_worker_nodes(scheduler)(*yaml_files['get_worker_nodes'])
+                job_ids, user_names, job_states, _ = get_jobs_info(scheduler)(*yaml_files['get_jobs_info'])
+                total_running_jobs, total_queued_jobs, qstatq_lod = get_queues_info(scheduler)(*yaml_files['get_queues_info'])
+                deprecate_old_yaml_files(*yaml_files['get_jobs_info'])
+                #  MAIN ##################################
+                logging.info('CALCULATION AREA')
+                cluster_dict, NAMED_WNS = calculate_cluster(worker_nodes)
+                workernodes_occupancy, cluster_dict = calculate_wn_occupancy(cluster_dict, user_names, job_states, job_ids)
+                h_stop = config['h_stop'] if h_stop is None else h_stop
+                v_stop = config['v_stop'] if v_stop is None else v_stop
+
+                display_parts = {
+                    'job_accounting_summary': (display_job_accounting_summary, (cluster_dict, total_running_jobs, total_queued_jobs, qstatq_lod)),
+                    'workernodes_matrix': (display_wn_occupancy, (workernodes_occupancy, cluster_dict)),
+                    'user_accounts_pool_mappings': (display_user_accounts_pool_mappings, (workernodes_occupancy['account_jobs_table'], workernodes_occupancy['pattern_of_id']))
+                }
+                logging.info('DISPLAY AREA')
+
+                print "\033c"
+
+                for idx, part in enumerate(config['user_display_parts'], 1):
+                    display_func, args = display_parts[part][0], display_parts[part][1]
+                    display_func(*args) if not sections_off[idx] else None
+                print "\nLog file created in %s" % expandvars(QTOP_LOGFILE)
+                sys.stdout.flush()
+                sys.stdout.close()
+                sys.stdout = stdout  # sys.stdout is back to its normal function (i.e. screen output)
+
+                num_lines = sum(1 for line in open(fout, 'r')) if not num_lines else num_lines
+                ansi_escape = re.compile(r'\x1b[^m]*m')
+                max_line_len = max(len(ansi_escape.sub('', line.strip())) for line in open(fout, 'r')) \
+                    if not max_line_len else max_line_len
+
+                logging.debug('Total nr of lines: %s' % num_lines)
+                logging.debug('Max line length: %s' % max_line_len)
+
+                line_offset = v_stop - v_start
+                cat_command = 'clear;tail -n+%s %s | head -n%s' % (v_start, fout, line_offset)
+                NOT_FOUND = subprocess.call(cat_command, stdout=stdout, stderr=stdout, shell=True)
+                # justification for implementation: http://unix.stackexchange.com/questions/47407/cat-line-x-to-line-y-on-a-huge
+                # -file
```

Comment:
gee ... URL split across 2 lines? please no :-( also, let's have it a couple of lines higher, *before* the line that matters...